classes/02class/exercises/c02-network07/mfreitassm/README.md

```diff
@@ -0,0 +1,393 @@
+            "VpcId": "vpc-04426efcf5b2228a4",
+            "OwnerId": "490308117505"
+        },
+        {
+            "Associations": [
+                {
+                    "Main": true,
+                    "RouteTableAssociationId": "rtbassoc-b73c61ce",
+                    "RouteTableId": "rtb-90ea79f6",
+                    "AssociationState": {
+                        "State": "associated"
+                    }
+                }
+            ],
+            "PropagatingVgws": [],
+            "RouteTableId": "rtb-90ea79f6",
+            "Routes": [
+                {
+                    "DestinationCidrBlock": "172.31.0.0/16",
+                    "GatewayId": "local",
+                    "Origin": "CreateRouteTable",
+                    "State": "active"
+                },
+                {
+                    "DestinationCidrBlock": "0.0.0.0/0",
+                    "GatewayId": "igw-44a1be20",
+                    "Origin": "CreateRoute",
+                    "State": "active"
+                }
+            ],
+            "Tags": [],
+            "VpcId": "vpc-ac525bcb",
+            "OwnerId": "490308117505"
+        }
+    ]
+}
+```
+
+- Answer the following questions:
+  - How did you accomplish both instructions?
+    ```
+    Using thoses commands above  by AWS CLI.
+    ```
+
+  - Why did you add 0.0.0.0 route to the IGW on public route table?
+    ```
+    0.0.0.0/0 represents all possible IP addresses. In the context of the way routing tables get set up by default on AWS, 0.0.0.0/0 is effectively "all non local addresses". This is because another route presumably exists in the routing table to route the VPC subnet to the local network on the VPC.
+    ```
+
+  - Why did you add 0.0.0.0 route to the NGW on private route table?
+    ```
+    0.0.0.0/0 represents all possible IP addresses. In the context of the way routing tables get set up by default on AWS, 0.0.0.0/0 is effectively "all non local addresses". This is because another route presumably exists in the routing table to route the VPC subnet to the local network on the VPC.
+    ```
+    
+  - What is the difference of IGW and NGW?
+    ```
+	A NAT device forwards traffic from the instances in the private subnet to the internet or other AWS services, and then sends the response back to the instances. It only works one way. The internet at large cannot get through your NAT to your private resources unless you explicitly allow it. While Internet Gateway is used to allow resources in your VPC to access internet. It is not a physical device. Only one can be associated with each VPC.
+    ```
+    
+  - Can you delete the destination route to your VPC network? Why?
+    ```
+    No, because it's a local route and if you delete it then the local machines can't connect to each other in AWS.
+    ```
+    
+  - What happens if you do not associate your route table with any subnets?
+    ```
+    it will automatically associate with a main route table for the VPC.
```

Comment:
This needs changing as your new route table will not be associated with anything.

There is a route table that is created with the VPC which is the default (main). Any subnets that are not associated with a route table will be automatically associated with that "main" route table.

So any subnets that are not associated with your new route table will be left on the "main" route table.

If you do not associate any subnets to the 2 new route tables, they will just stand there unused (because they are not the main route table)

Reply:
Actually, I changed my mind I can understand your last answer: the subnets will be associated with the main one as you said.